server/planning/events.py

```diff
@@ -435,3 +441,14 @@ def generate_recurring_dates(start, frequency, interval=1, until=None, byday=Non
         return (tz.localize(dt).astimezone(pytz.UTC).replace(tzinfo=None) for dt in dates)
     else:
         return (date for date in dates)
+
+
+def setRecurringMode(event):
```

Comment:
For the first level fields I agree that you have to reset from the client.
But on eve if you patch a field of type dictionary you expect that the old dictionary will be replaced with the new one.This is happening for example if you patch a field of type list the old one will be replaced with the new one.
On dictionary it will perform a deep update and will not replace the entire dictionary.
I think this is not a common patch behavior.